tests/unit/TestNetRemoteDataStreamingReactors.cxx

```diff
@@ -70,3 +70,62 @@ DataStreamWriter::NextWrite()
         StartWritesDone();
     }
 }
+
+DataStreamReader::DataStreamReader(NetRemoteDataStreaming::Stub* client, DataStreamDownloadRequest* request)
+{
+    client->async()->DataStreamDownload(&m_clientContext, request, this);
+    StartCall();
+    StartRead(&m_data);
+}
+
+void
+DataStreamReader::OnReadDone(bool isOk)
+{
+    if (isOk) {
+        m_numberOfDataBlocksReceived++;
+        StartRead(&m_data);
+    }
+    // If read fails, then there is likely no more data to be read, so do nothing.
+}
+
+void
+DataStreamReader::OnDone(const grpc::Status& status)
+{
+    std::unique_lock lock(m_readStatusGate);
+
+    m_status = status;
+    m_done = true;
+    m_readsDone.notify_one();
+}
+
+grpc::Status
+DataStreamReader::Await(uint32_t* numberOfDataBlocksReceived, DataStreamOperationStatus* operationStatus)
+{
+    std::unique_lock lock(m_readStatusGate);
+    static constexpr auto timeoutValue = 10s;
```

Comment:
I think we'll want to eventually use these test classes outside test code (and move them to a common lib at that time).

In that case, it's likely we'll want to make this timeout value configurable. Suggest to at least add a member to `DataStreamMember` and read the member here instead of using the constant. Completely fine if the member is just initialized to a constant (with this value) and no way is provided for configuring it for now.